drv/ksz8463/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+    MACAR1 = register_offset(0x010),
+    MACAR2 = register_offset(0x012),
+    MACAR3 = register_offset(0x014),
+
+    P1MBCR = register_offset(0x04c),
+    P1MBSR = register_offset(0x04e),
+
+    P2MBCR = register_offset(0x058),
+    P2MBSR = register_offset(0x05a),
+
+    CFGR = register_offset(0x0d8),
+    DSP_CNTRL_6 = register_offset(0x734),
+}
+
+pub struct Ksz8463 {
+    spi: SpiDevice,
+    nrst: PinSet,
+    slow_reset: bool,
+}
+
+impl Ksz8463 {
+    pub fn new(spi: SpiDevice, nrst: PinSet, slow_reset: bool) -> Self {
+        Self {
+            spi,
+            nrst,
+            slow_reset,
+        }
+    }
+
+    pub fn read(&self, r: Register) -> Result<u16, SpiError> {
+        let cmd = (r as u16).to_be_bytes();
+        let request = [cmd[0], cmd[1]];
+        let mut response = [0; 4];
+
+        self.spi.exchange(&request, &mut response)?;
+        let v = u16::from_le_bytes(response[2..].try_into().unwrap());
+        ringbuf_entry!(Trace::Read(r, v));
+
+        Ok(v)
+    }
+
+    pub fn write(&self, r: Register, v: u16) -> Result<(), SpiError> {
+        let cmd = (r as u16 | 0x8000).to_be_bytes(); // Set MSB to indicate write.
+        let data = v.to_le_bytes();
+        let request = [cmd[0], cmd[1], data[0], data[1]];
+
+        ringbuf_entry!(Trace::Write(r, v));
+        self.spi.write(&request[..])?;
+        Ok(())
+    }
+
+    pub fn write_masked(
+        &self,
+        r: Register,
+        v: u16,
+        mask: u16,
+    ) -> Result<(), SpiError> {
+        let _v = (self.read(r)? & !mask) | (v & mask);
+        self.write(r, _v)
+    }
+
+    pub fn enabled(&self) -> Result<bool, SpiError> {
+        Ok(self.read(Register::CIDER)? & 0x1 != 0)
+    }
+
+    pub fn enable(&self) -> Result<(), SpiError> {
+        self.write(Register::CIDER, 1)
+    }
+
+    pub fn disable(&self) -> Result<(), SpiError> {
+        self.write(Register::CIDER, 0)
+    }
+
+    /// Configures the KSZ8463 switch in 100BASE-FX mode.
+    pub fn configure(&self) {
+        let gpio_driver = GPIO.get_task_id();
+        let gpio_driver = Gpio::from(gpio_driver);
+
+        gpio_driver.reset(self.nrst).unwrap();
+        gpio_driver
+            .configure_output(
+                self.nrst,
+                OutputType::PushPull,
+                Speed::Low,
+                Pull::None,
+            )
+            .unwrap();
+
+        // Toggle the reset line
+        sleep_for(10); // Reset must be held low for 10 ms after power up
+        gpio_driver.set(self.nrst).unwrap();
+
+        // The datasheet recommends a particular combination of diodes and
+        // capacitors which dramatically slow down the rise of the reset
+        // line, meaning you have to wait for extra long here.
+        //
+        // Otherwise, the minimum wait time is 1 µs, so 1 ms is fine.
+        sleep_for(if self.slow_reset { 150 } else { 1 });
+
+        let id = self.read(Register::CIDER).unwrap();
+        assert_eq!(id & !1, 0x8452);
```

Comment:
Will panicking the task that's calling this function fix the communication with the switch? I'd suggest making this an `Err` return and letting the caller decide.

Reply:
We talked about this in person and decided that panicking is okay, since this is only called at initialization and rebooting the task is a reasonable response to such a failure.